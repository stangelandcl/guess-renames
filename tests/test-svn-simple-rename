#!/bin/bash

mkdir testdir
pushd testdir > /dev/null

echo "*** Creating repository."
svnadmin create test-repo
svn co file:///$PWD/test-repo test-wc
pushd test-wc > /dev/null
echo "*** Repository created."

(echo "alpha" && echo "bravo" && echo "charlie") > 1
(echo "delta" && echo "echo" && echo "foxtrot") > 2

echo "*** Files created. Contents:"

echo "*** 1:"
cat 1
echo "*** 2:"
cat 2

echo "*** Adding files."

svn st
svn add *
svn st
svn commit -m "Adding"

echo "*** Files added."

echo "*** Renaming."

mv 1 3
mv 2 4

echo "*** Renamed. Contents:"

echo "*** 3:"
cat 3
echo "*** 4:"
cat 4

echo "*** Guessing."

svn st
guess-renames
svn st
svn commit -m "Renaming"
svn st

echo "*** Done."

popd > /dev/null # test-wc
popd > /dev/null # test-repo
rm -rf testdir
exit 0