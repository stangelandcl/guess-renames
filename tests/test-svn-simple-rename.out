*** Creating repository.
Checked out revision 0.
*** Repository created.
*** Files created. Contents:
*** 1:
alpha
bravo
charlie
*** 2:
delta
echo
foxtrot
*** Adding files.
?       1
?       2
A         1
A         2
A       1
A       2
Adding         1
Adding         2
Transmitting file data ..
Committed revision 1.
*** Files added.
*** Renaming.
*** Renamed. Contents:
*** 3:
alpha
bravo
charlie
*** 4:
delta
echo
foxtrot
*** Guessing.
?       3
?       4
!       1
!       2
2 is now 4 (score: 3.000000)
1 is now 3 (score: 3.000000)
D       1
D       2
A  +    3
A  +    4
Deleting       1
Deleting       2
Adding         3
Adding         4

Committed revision 2.
*** Done.
